components/studio/bin/hab-studio.sh

```diff
@@ -356,17 +338,22 @@ subcommand_run() {
   shift "$((OPTIND - 1))"
 
   new_studio
-  run_studio $*
+  run_studio "$@"
 }
 
-# **Internal** Creates a new Studio.
-new_studio() {
-  # Check if a pre-existing Studio configuration is found and use that to
-  # determine the type
+# **Internal** Check if a pre-existing Studio configuration is found and use
+# that to determine the type and assign it to STUDIO_TYPE
+source_studio_type_config() {
   if [ -s "$studio_config" ]; then
+    # shellcheck disable=1090
     . "$studio_config"
```

Comment:
It turns out [`hab-studio-type-default.sh`](https://github.com/habitat-sh/habitat/blob/master/components/studio/libexec/hab-studio-type-default.sh#L6) references a positional parameter, which means that we have to pass those args into `source_studio_type_config` or things will very subtly break. I think we should look into changing that as I think it's a fairly brittle and confusing abstraction.

Reply:
I made some changes to try to evaluate this later and give an error at the more appropriate time. See:
1. https://github.com/habitat-sh/habitat/pull/4661/files#diff-b102b5e067962e9cce22e9e8d3541fefR591
2. https://github.com/habitat-sh/habitat/pull/4661/files#diff-b102b5e067962e9cce22e9e8d3541fefR727
3. https://github.com/habitat-sh/habitat/pull/4661/files#diff-98ec64e3f381a5ccde4f4506db6aae9cR6

Though maybe we should just rework this so that `studio_build_command` doesn't need to reference `$1`, as that's super confusing.

Reply:
This is moot now; I reworked it to be simpler: https://github.com/habitat-sh/habitat/pull/4661/commits/ae4e0d69078762aadaa685d3bbe03686e7a9e4c8